mllib/src/main/scala/org/apache/spark/mllib/util/LinearDataGenerator.scala

```diff
@@ -112,8 +139,13 @@ object LinearDataGenerator {
     x.foreach { v =>
       var i = 0
       val len = v.length
+      val sparceRnd = new Random(seed)
```

Comment:
Since you seed `rnd` and `sparceRnd` with the same seed, both of them will generate the same sequence of random numbers which is not what you want. You should be able to use the same random number generator which will give you uncorrelated random numbers in both creating the features and choice which columns to zero out.